### PR TITLE
Font family preview in the font family picker

### LINKED ISSUE #51151: Font Picker: Refine, show font preview, and load only active fonts from library
Extracted from #45271.

The font picker will show fonts that are registered in theme.json. As detailed in #45271, that means fonts added to your library which are _active_.  The font picker can look like this:

Selecting fonts in Post Editor

The above design is an enhancement that focuses on showing a preview of the font itself, inline in the picker. But this design should not block either the webfonts API from landing, nor the font manager. The existing picker can work fine in the same system:

existing picker

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Style font family option in the font family picker.

## Why?
To help users to select the desired font.
Fixes: https://github.com/WordPress/gutenberg/issues/51151

## How?
Using the [CustomSelectControl](https://wordpress.github.io/gutenberg/?path=/docs/components-customselectcontrol--docs) component and setting the right style.

## Testing Instructions
Open a font family picker and verify that each font family renders the family's name using the family's typography.

## Screenshot

| Before | After |
| --- | --- |
| image | image |

### CODE REVIEW

**Review by tyxla — COMMENTED** [review]
Delaying the review as feedback was provided in the related issue: https://github.com/WordPress/gutenberg/issues/51151#issuecomment-2487006494

**Comment by matiasbenedetto** [comment]
I updated the PR using the [CustomSelectControl](https://wordpress.github.io/gutenberg/?path=/docs/components-customselectcontrol--docs) as suggested by @mirka in https://github.com/WordPress/gutenberg/issues/51151#issuecomment-2487006494.
It seems to be ready for review.

**Comment by annezazu** [comment]
Thanks for moving this forward! I just tried to test this PR in WordPress playground and found that, while I could now see the Font Families, they didn't seem to properly update across the site, both when I made changes globally and to an individual block:

https://github.com/user-attachments/assets/b26b3662-3532-4712-9cd7-e73f57412e88

I'm not sure what's going on there. It might be Playground or something is off :) If it's playground, I'll report it to them.

File: packages/block-editor/src/components/font-family/index.js
Comment (tyxla) [inline packages/block-editor/src/components/font-family/index.js]: I think you need to remove this prop.

File: packages/block-editor/src/components/font-family/index.js
Comment (tyxla) [inline packages/block-editor/src/components/font-family/index.js]: I think this prop also should be removed. Aren't you getting some errors in the console when passing this prop? I think you should, since it will end up being wrongly passed down to the underlying DOM element.

File: packages/block-editor/src/components/font-family/index.js
Comment (tyxla) [inline packages/block-editor/src/components/font-family/index.js]: This will yield an error right now. We need to set the right value, something like:

```suggestion
			onChange={ ( { selectedItem } ) => onChange( selectedItem.style.fontFamily ) }
```

**Review by tyxla — CHANGES_REQUESTED** [review]
Haven't tested, but I expect this will currently trigger a bunch of errors while rendering and while setting a new item. This is likely what @annezazu experienced. 

Left a few suggestions that should resolve all issues.

I'd recommend keeping an eye on the console to ensure things work correctly and there are no errors.

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: It looks like we are passing the `options` as described in the component docs. See the `options` description and data example in storybook: https://wordpress.github.io/gutenberg/?path=/docs/components-customselectcontrol--docs

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: Not sure what prop are you pointing to.

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: I don't see console errors and it is used like that in other parts of the codebase. Example:
https://github.com/WordPress/gutenberg/blob/bc8c4a6804686bf23779f7924dd5962e667feac6/packages/block-editor/src/components/font-appearance-control/index.js#L158

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: If you mean `__next40pxDefaultSize`, I see it used in other parts of the codebase where CustomSelectControl is rendered, too:
https://github.com/WordPress/gutenberg/blob/bc8c4a6804686bf23779f7924dd5962e667feac6/packages/block-editor/src/components/font-appearance-control/index.js#L155

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: Thanks for pointing this, fixed here: https://github.com/WordPress/gutenberg/pull/67118/commits/c39b0f1051a88f8298fba281e1d5877246973904

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: Maybe you are referring something that is valid for the `CustomSelectControlV2` ? I saw that control in the codebase, but it's not listed on [storybook](https://wordpress.github.io/gutenberg/?path=/docs/docs-introduction--page), and it doesn't seem to be in use in the Gutenberg codebase, so I'm not sure what it's their current state.

**Comment by matiasbenedetto** [comment]
Thanks for the reviews! With the latest change in `onChange` , it seems to be working as expected:

https://github.com/user-attachments/assets/be2b0609-114f-4cb5-a610-1097ad2cec4c

File: packages/block-editor/src/components/font-family/index.js
Comment (tyxla) [reply packages/block-editor/src/components/font-family/index.js]: My comment is referring to the `labelPosition` prop, not the `options` prop.

File: packages/block-editor/src/components/font-family/index.js
Comment (tyxla) [reply packages/block-editor/src/components/font-family/index.js]: My comment is referring to the `__nextHasNoMarginBottom` prop, actually.

**Comment by matiasbenedetto** [comment]
> I just tried to test this PR in WordPress playground and found that, while I could now see the Font Families

@annezazu Thanks again for testing! I tried it in Playground, and it's  working fine now:


https://github.com/user-attachments/assets/d97aa3ae-9c07-4a20-8560-46c90dac907f

**Review by tyxla — COMMENTED** [review]
Almost ready to go, we just need to remove the two props (see my answers to your questions in the preview review comments)

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: ok, removed https://github.com/WordPress/gutenberg/pull/67118/commits/1f94e6cc43af38933ce5810f30e617db26143210

File: packages/block-editor/src/components/font-family/index.js
Comment (matiasbenedetto) [reply packages/block-editor/src/components/font-family/index.js]: removed here: https://github.com/WordPress/gutenberg/pull/67118/commits/1f94e6cc43af38933ce5810f30e617db26143210

**Comment by matiasbenedetto** [comment]
> we just need to remove the two props

Thanks for the reviews. Those 2 props were removed in this commit: https://github.com/WordPress/gutenberg/pull/67118/commits/1f94e6cc43af38933ce5810f30e617db26143210

File: test/e2e/specs/site-editor/font-library.spec.js
Comment (mirka) [inline test/e2e/specs/site-editor/font-library.spec.js]:
```suggestion
			await page.getByRole( 'combobox', { name: 'Font' } ).click();
```

Looking at the rendered HTML, it looks like this should be `role="combobox"`?

**Review by mirka — APPROVED** [review]
Tests well without console errors 👍 I think we're good to go once the tests pass.

File: test/e2e/specs/site-editor/font-library.spec.js
Comment (matiasbenedetto) [reply test/e2e/specs/site-editor/font-library.spec.js]: Thanks for taking a look. I'm looking at how to fix the test locally, but I'm having some difficulties running the tests properly in my local environment.

Comment (matiasbenedetto) [reply test/e2e/specs/site-editor/font-library.spec.js]: @jeryj fixed it here: https://github.com/WordPress/gutenberg/pull/67118/commits/a10e2e9bb970b73e946bb9155d5b74dbd994ab2a

**Comment by matiasbenedetto** [comment]
@tyxla, your review is blocking the merge of this PR. Could you please take another look or lift the block? Thanks!

**Review by tyxla — APPROVED** [review]
LGTM, thanks @matiasbenedetto 🚀

**Pull request merged**
